File: data/imported/knowledge/slides/net/merged/snippets/snippet_028.cs
[Fact]
    public void TestGlow()
    {
        // Glow effect persists.
        using var pres = new Presentation();
        var slide = pres.Slides[0];
        slide.Shapes!.Clear();
        var shape = slide.Shapes.AddAutoShape(ShapeType.Ellipse, 100, 100, 200, 200);
        var ef = shape.EffectFormat;
        ef.EnableGlowEffect();
        ef.GlowEffect!.Radius = 15;
        ef.GlowEffect.Color.Color = Color.Gold;

        using var pres2 = TestHelpers.SaveAndReopen(pres, _tempDir);
        var g2 = pres2.Slides[0].Shapes![0].EffectFormat.GlowEffect;
        g2.Should().NotBeNull("glow_effect should not be None after reload");
        g2!.Radius.Should().Be(15);
    }

File: data/imported/knowledge/slides/java/merged/snippets/snippet_028.java
@Test
    void testOuterShadow() throws IOException {
        try (var pres = new Presentation()) {
            var slide = clear(pres);
            var shape = slide.getShapes().addAutoShape(ShapeType.RECTANGLE, 100, 100, 200, 100);
            var ef = shape.getEffectFormat();
            ef.enableOuterShadowEffect();
            var shadow = ef.getOuterShadowEffect();
            shadow.setBlurRadius(10);
            shadow.setDirection(315);
            shadow.setDistance(8);
            shadow.getShadowColor().setColor(Color.fromArgb(128, 0, 0, 0));

            try (var pres2 = saveAndReopen(pres)) {
                var ef2 = pres2.getSlides().get(0).getShapes().get(0).getEffectFormat();
                var s2 = ef2.getOuterShadowEffect();
                assertThat(s2).as("outer_shadow_effect should not be None after reload").isNotNull();
                assertThat(s2.getBlurRadius()).isEqualTo(10);
                assertThat(s2.getDirection()).isEqualTo(315);
                assertThat(s2.getDistance()).isEqualTo(8);
            }
        }
    }